# MMM
Mercury Methylating Microbes in Vermont
PRJNA 277296
Code for amplicon data processing and analysis for this manuscript in prep can be found here. Raw amplicon data are stored at the NCBI SRA under BioProject PRJNA 1277296

**Microbial Mercury Methylation in Winooski River and Englesby Brook, Vermont **
H. Sophia Fatima1, Sam Byrne1,2, Peter Ryan3, Erin Eggleston1,*
1   Middlebury College, Biology Department,Middlebury, Vermont, USA
2   Middlebury College, Global Health Program, Middlebury, Vermont, USA
3   Middlebury College, Earth and Climate Science Department,Middlebury, Vermont, USA

*   Correspondence: eeggleston@middlebury.edu; Tel.: 802-443-5779

Abstract

Mercury methylation is the harmful process by which mercury (Hg) is converted to methylmercury (MeHg). Biotically, this is done by mercury-methylating microbes such as sulfate-reducing bacteria (SRB) in ecosystems such as wetlands and aquatic sediments. Due to legacy Hg and sulfate deposition along the eastern US, there likely remains a moderate concentration of Hg in Winooski River (W) and Englesby Brook (E), sites which have been identified as having relatively higher mercury concentrations in Vermont. If the microbial community consists partially of mercury-methylating microbes, there is a higher potential of conversion of Hg to MeHg. To investigate this potential, sediment cores were taken from the sites to analyze microbial community composition, the presence of hgcA genes within sample metagenomes, and concentrations of Hg, C, and N. Results suggest that the microbial communities of Winooski River and Englesby Brook are different. Further, the Riparian Land samples had a significantly greater relative abundance of 16S rRNA ASVs that matched to Hg-methylating genera compared to Riparian Water samples, with greater THg concentrations in Englesby Brook. hgcA genes were present within samples from all locations and sampling sites, although Winooski Riparian Water and Winooski Bay had the greatest number of amino acid sequence hits.

Keywords: microbial mercury methylation, community composition, river sediments, Vermont, CN analysis, sulfate-reducing bacteria, 16S rRNA, metagenome
